op-node/rollup/derive/engine_consolidate.go

```diff
@@ -88,21 +89,34 @@ func logL1InfoTxns(l log.Logger, l2Number, l2Timestamp uint64, safeTx, unsafeTx
 	}
 
 	// Then decode the ABI encoded parameters
-	var safeInfo, unsafeInfo L1BlockInfo
-	errSafe = (&safeInfo).UnmarshalBinary(safeTxValue.Data())
-	errUnsafe = (&unsafeInfo).UnmarshalBinary(unsafeTxValue.Data())
+	safeInfo, errSafe := L1BlockInfoFromBytes(rollupCfg, l2Timestamp, safeTxValue.Data())
+	unsafeInfo, errUnsafe := L1BlockInfoFromBytes(rollupCfg, l2Timestamp, unsafeTxValue.Data())
 	if errSafe != nil || errUnsafe != nil {
 		l.Error("failed to umarshal l1 info", "errSafe", errSafe, "errUnsafe", errUnsafe)
 		return
 	}
 
-	l.Error("L1 Info transaction differs", "number", l2Number, "time", l2Timestamp,
-		"safe_l1_number", safeInfo.Number, "safe_l1_hash", safeInfo.BlockHash,
-		"safe_l1_time", safeInfo.Time, "safe_seq_num", safeInfo.SequenceNumber,
-		"safe_l1_basefee", safeInfo.BaseFee, "safe_batcher_add", safeInfo.BlockHash,
-		"safe_gpo_scalar", safeInfo.L1FeeScalar, "safe_gpo_overhead", safeInfo.L1FeeOverhead,
-		"unsafe_l1_number", unsafeInfo.Number, "unsafe_l1_hash", unsafeInfo.BlockHash,
-		"unsafe_l1_time", unsafeInfo.Time, "unsafe_seq_num", unsafeInfo.SequenceNumber,
-		"unsafe_l1_basefee", unsafeInfo.BaseFee, "unsafe_batcher_add", unsafeInfo.BlockHash,
-		"unsafe_gpo_scalar", unsafeInfo.L1FeeScalar, "unsafe_gpo_overhead", unsafeInfo.L1FeeOverhead)
+	if bytes.HasPrefix(safeTxValue.Data(), types.EcotoneL1AttributesSelector) {
```

Comment:
This if/else is really hard to read, perhaps better to have 2 functions that log or reuse more code somehow